### PR TITLE
Enhancements to Security docs Basic authentication tutorial

### PR DESCRIPTION
**Pull request by MichalMaler**
Provided review to this section as per description in https://issues.redhat.com/browse/QDOCS-95

Signed-off-by: Michal Maléř <mmaler@redhat.com>

### CODE REVIEW

**Comment by MichalMaler** [comment]
Hello folks. I prepared a small PR that is ready to be merged if reviewed by on of you and approve by Upstream community.

@sheilamjones @michelle-purcell @jmartisk @sberyozkin

File: docs/src/main/asciidoc/security-basic-authentication-tutorial.adoc [hunk 1]
```diff
@@ -14,15 +14,15 @@ include::{includes}/prerequisites.adoc[]
...
 
 [cols="20%,40% ",options="header"]
 |===
 |Endpoint | Description 
 |`/api/public`| The `/api/public` endpoint can be accessed anonymously.
 |`/api/admin`| The `/api/admin` endpoint is protected with role-based access control (RBAC), and only users who have been granted the `admin` role can access it.
-At this endpoint, the `@RolesAllowed` annotation is used to declaratively enforce the access constraint.
-|`/api/users/me`| The `/api/users/me` endpoint is protected with RBAC and only users that have been granted the `user` role can access it. A JSON document with details about the user is returned as a response.
+At this endpoint, the `@RolesAllowed` annotation is used to enforce the access constraint declaratively.
+|`/api/users/me`| The `/api/users/me` endpoint is protected with RBAC and only users that have been granted the `user` role can access it.
```
Comment (jmartisk) [inline docs/src/main/asciidoc/security-basic-authentication-tutorial.adoc:25]: You're missing the sentence about what it returns (it should be just a string with the username - not a JSON document)

File: docs/src/main/asciidoc/security-basic-authentication-tutorial.adoc [hunk 2]
```diff
@@ -537,8 +538,8 @@ public class Role extends PanacheEntity {
 By default, passwords are stored and hashed by using https://en.wikipedia.org/wiki/Bcrypt[bcrypt] under the 
 https://en.wikipedia.org/wiki/Crypt_(C)[Modular Crypt Format] (MCF).
 
-When you need to create a hashed password we provide the convenient `String BcryptUtil.bcryptHash(String password)`
-function, which defaults to creating a random salt and hashing in 10 iterations. 
+When you need to create a hashed password, provide the convenient `String BcryptUtil.bcryptHash(String password)`
```
Comment (jmartisk) [inline docs/src/main/asciidoc/security-basic-authentication-tutorial.adoc:541]: No, the "we" means Quarkus - the user doesn't have to provide it. 
The original meaning was correct, maybe to improve I would change "we provide" to "you may use"

**Review by jmartisk — CHANGES_REQUESTED** [review]

File: docs/src/main/asciidoc/security-basic-authentication-tutorial.adoc (hunk 1 above)
Comment (MichalMaler) [reply docs/src/main/asciidoc/security-basic-authentication-tutorial.adoc:25]: Yeah, I just add it. THX :)

File: docs/src/main/asciidoc/security-basic-authentication-tutorial.adoc (hunk 2 above)
Comment (MichalMaler) [reply docs/src/main/asciidoc/security-basic-authentication-tutorial.adoc:542]: OK, applying

**Comment by MichalMaler** [comment]
Will squash and rebase after reviews are done.

**Review by jmartisk — APPROVED** [review]

File: docs/src/main/asciidoc/security-basic-authentication-tutorial.adoc [hunk 3]
```diff
@@ -14,15 +14,16 @@ include::{includes}/prerequisites.adoc[]
...
+The steps in this tutorial guide you through building an application that provides the following endpoints to demonstrate different authorization policies:
 
 [cols="20%,40% ",options="header"]
 |===
 |Endpoint | Description 
 |`/api/public`| The `/api/public` endpoint can be accessed anonymously.
 |`/api/admin`| The `/api/admin` endpoint is protected with role-based access control (RBAC), and only users who have been granted the `admin` role can access it.
-At this endpoint, the `@RolesAllowed` annotation is used to declaratively enforce the access constraint.
-|`/api/users/me`| The `/api/users/me` endpoint is protected with RBAC and only users that have been granted the `user` role can access it. A JSON document with details about the user is returned as a response.
+At this endpoint, the `@RolesAllowed` annotation is used to enforce the access constraint declaratively.
```
Comment (rolfedh) [inline docs/src/main/asciidoc/security-basic-authentication-tutorial.adoc:24]: Maybe 
```suggestion
At this endpoint, the `@RolesAllowed` annotation enforces the access constraint declaratively.
```

File: docs/src/main/asciidoc/security-basic-authentication-tutorial.adoc [hunk 4]
```diff
@@ -14,15 +14,16 @@ include::{includes}/prerequisites.adoc[]
...
+|`/api/users/me`| The `/api/users/me` endpoint is protected with RBAC and only users that have been granted the `user` role can access it.
```
Comment (rolfedh) [inline docs/src/main/asciidoc/security-basic-authentication-tutorial.adoc:25]: Maybe 
```suggestion
|`/api/users/me`| The `/api/users/me` endpoint is protected by RBAC. Only users that have the `user` role can access the endpoint.
```

File: docs/src/main/asciidoc/security-basic-authentication-tutorial.adoc [hunk 5]
```diff
@@ -14,15 +14,16 @@ include::{includes}/prerequisites.adoc[]
...
+This endpoint returns a string value containing the caller's username.
```
Comment (rolfedh) [inline docs/src/main/asciidoc/security-basic-authentication-tutorial.adoc:26]: Maybe 
```suggestion
This endpoint returns the caller's username as a string.
```

File: docs/src/main/asciidoc/security-basic-authentication-tutorial.adoc [hunk 6]
```diff
@@ -203,8 +204,8 @@ public class User extends PanacheEntity {
 The `security-jpa` extension initializes only if there is a single entity annotated with `@UserDefinition`.
 
 <1> The `@UserDefinition` annotation must be present on a single entity and can be either a regular Hibernate ORM entity or a Hibernate ORM with Panache entity.
-<2> Indicates the field used for the user name.
-<3> Indicates the field used for the password, which defaults to using bcrypt hashed passwords but you can also configure it for plain text or custom passwords.
+<2> Indicates the field used for the username.
+<3> Indicates the field used for the password, which defaults to using bcrypt hashed passwords, but you can also configure it for plain text or custom passwords.
```
Comment (rolfedh) [inline docs/src/main/asciidoc/security-basic-authentication-tutorial.adoc:208]:
```suggestion
<3> Indicates the field used for the password. By default, it uses bcrypt-hashed passwords. You can configure it to use plain text or custom passwords.
```

**Review by rolfedh — APPROVED** [review]
LGTM. I offered some completely optional suggestions that might improve clarity and ease localization.

**Comment by MichalMaler** [comment]
@rolfedh Thank you for nice suggestions! Applying, squashing, rebasing.

**Review by sberyozkin — COMMENTED** [review]
@MichalMaler sorry for the delay, nice update, I have only one minor concern,  I've always tried to avoid a space between a word and ':',  was adding a space here intentional?

**Comment by jmartisk** [comment]
LGTM, please squash into one commit and then I think we're good to go

**Review by sberyozkin — APPROVED** [review]

**Pull request merged**